server/sso/index.js

```diff
@@ -3,124 +3,137 @@ const base64url = require('base64url');
 const cookie = require('cookie');
 const proxy = require('request');
 
-const config = {
+const CONFIG = {
   ssoAuth: {
-    host: 'www.ovh.com',
-    baseUrl: 'https://www.ovh.com/cgi-bin/crosslogin.cgi',
-    devLoginUrl: 'https://www.ovh.com/auth/requestDevLogin/',
+    eu: {
+      host: 'www.ovh.com',
+      baseUrl: 'https://www.ovh.com/cgi-bin/crosslogin.cgi',
+      devLoginUrl: 'https://www.ovh.com/auth/requestDevLogin/',
+    },
+    ca: {
+      host: 'ca.ovh.com',
+      devLoginUrl: 'https://ca.ovh.com/auth/requestDevLogin/',
+      baseUrl: 'https://ca.ovh.com/cgi-bin/crosslogin.cgi',
+    },
+    us: {
+      host: 'ovhcloud.com',
+      devLoginUrl: 'https://us.ovhcloud.com/auth/requestDevLogin/',
+      baseUrl: 'https://us.ovhcloud.com/cgi-bin/crosslogin.cgi',
+    },
   },
 };
 
-// [SSO] authentication
-function login(req, res) {
-  console.log('[SSO] - crosslogin');
-
-  const { headers } = req;
-  headers.host = config.ssoAuth.host;
-
-  const location = new Promise((resolve) => {
-    proxy.get({
-      url: config.ssoAuth.baseUrl + req.url,
-      headers,
-      proxy: config.proxy ? config.proxy.host : '',
-      followRedirect: false,
-    }, (err, resp) => {
-      if (err) {
-        console.error('[SSO] - crosslogin - error: ', err);
-        return resp.status(500);
-      }
-
-      const cookies = resp.headers['set-cookie'];
-      let parsedCookie;
-
-      for (let i = cookies.length - 1; i >= 0; i -= 1) {
-        parsedCookie = cookie.parse(cookies[i]);
+class Sso {
+  constructor(region) {
+    this.config = CONFIG.ssoAuth[region];
+  }
 
-        if (parsedCookie['CA.OVH.SES']) {
-          res.cookie('CA.OVH.SES', parsedCookie['CA.OVH.SES'], { path: '/', httpOnly: true });
-        }
-        if (parsedCookie.SESSION) {
-          res.cookie('SESSION', parsedCookie.SESSION, { path: '/', httpOnly: true });
+  login(req, res) {
+    console.log('[SSO] - crosslogin');
+
+    const { headers } = req;
+    headers.host = this.config.host;
+
+    const location = new Promise((resolve) => {
+      proxy.get({
+        url: this.config.baseUrl + req.url,
+        headers,
+        proxy: this.config.proxy ? this.config.proxy.host : '',
+        followRedirect: false,
+      }, (err, resp) => {
+        if (err) {
+          console.error('[SSO] - crosslogin - error: ', err);
+          return resp.status(500);
         }
-        if (parsedCookie.USERID) {
-          res.cookie('USERID', parsedCookie.USERID, { path: '/' });
+
+        const cookies = resp.headers['set-cookie'];
+        let parsedCookie;
+
+        for (let i = cookies.length - 1; i >= 0; i -= 1) {
```

Comment:
Can you maybe `cookies.reverse().map(cookie => ...)` ?

Reply:
cookies.reverse().forEach(...) :p